### PR TITLE
Detox Website: initial version.

### PR DESCRIPTION
**Pull request by asafkorem**
This pull request creates the initial version of Detox website (using [Docusaurus](http://docusaurus.io/)), along with GitHub action for deploying the website when `master` changes.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report---android.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ IMPORTANT: We will not be able to help out or provide proper analysis without th
 ### Device logs (adb logcat)
 <!--
 Paste *device* logs from the Android device/emulator - associated with the failing tests, under the "details" tag below.
-Device logs can be obtained either by using `adb` (e.g. by running `adb logcat` while the test is running), or by running having Detox generate them as test artifacts (i.e. by providing the `--record-logs all` argument to the "detox test ..." command - see artifacts guide (https://github.com/wix/Detox/blob/master/docs/APIRef.Artifacts.md) for more info).
+Device logs can be obtained either by using `adb` (e.g. by running `adb logcat` while the test is running), or by running having Detox generate them as test artifacts (i.e. by providing the `--record-logs all` argument to the "detox test ..." command - see artifacts guide (https://wix.github.io/detox/docs/api/artifacts) for more info).
```
Comment (noomorph) [inline .github/ISSUE_TEMPLATE/bug-report---android.md:52]: This is an error, Asaf. It should have been an internal link. Fixed in #3074

Comment (noomorph) [reply .github/ISSUE_TEMPLATE/bug-report---android.md:52]: Ah, my bad, not this file :D disregard meanwhile

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+name: documentation
```
Comment (noomorph) [inline .github/workflows/documentation.yml:1]: rename it to website, pls :)

Comment (asafkorem) [reply .github/workflows/documentation.yml:1]: v

File: website/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Website
+
+This website is built using [Docusaurus 2](https://docusaurus.io/), a modern static website generator.
```
Comment (noomorph) [inline website/README.md:3]: Do we really need this README on the main page? 🤔

File: website/src/pages/index.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      </div>
+    </header>
+  );
+}
+
+export default function Home() {
+  const {siteConfig} = useDocusaurusContext();
+  return (
+    <Layout
+      title={`Hello from ${siteConfig.title}`}
```
Comment (noomorph) [inline website/src/pages/index.js:35]: Is this not finished?

File: website/src/pages/markdown-page.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (noomorph) [inline website/src/pages/markdown-page.md:1]: Isn't it redundant?

**Comment by noomorph** [comment]
@asafkorem  as for the images – could you explicitly list their URLs (here at least), where do they come from, and what their respective licenses are? We just need to keep in mind that we're a public project and need to comply with the copyright policies. Let's be 100% transparent about the assets.

File: docs/APIRef.Artifacts.md [hunk 6]
```diff
@@ -13,7 +20,7 @@ Artifacts are disabled by default. To enable them, specify via **launch argument
 * To record `.dtxrec` (Detox Instruments recordings) for each test, add `--record-performance all`. To open those recordings, you'll need [Detox Instruments](https://github.com/wix/DetoxInstruments). **NOTE:** only iOS is supported.
 * To capture `.uihierarchy` snapshots (**iOS only, Xcode 12.0+**) on view action failures, add `--capture-view-hierarchy enabled`.
 * To take `.png` screenshots before and after each test, add `--take-screenshots all` (or `--take-screenshots failing`, if you want to keep only screenshots of failing tests).  
-Alternatively, you might leverage the [device.takeScreenshot()](APIRef.DeviceObjectAPI.md#devicetakescreenshotname) API for manual control.
+Alternatively, you might leverage the [device.takeScreenshot()](https://wix.github.io/docs/api/artifacts#devicetakescreenshotname) API for manual control.
```
Comment (noomorph) [inline docs/APIRef.Artifacts.md:23]: @asafkorem, I am strongly against replacing the internal links with the external ones here in the documentation folder. That will make harder the whole process of linting the docs and finding broken anchors inside a specific Git revision. Please don't. I wholeheartedly ask you.

**Review by noomorph — CHANGES_REQUESTED** [review]

**Comment by noomorph** [comment]
Last but not least, @asafkorem do you have anywhere a local deployment we can go and take a look at?

**Comment by asafkorem** [comment]
> Last but not least, @asafkorem do you have anywhere a local deployment we can go and take a look at?

Sure, you can checkout this branch from my repository and execute `npm run start` from the website's dir.

File: website/README.md (hunk 3 above)
Comment (noomorph) [reply website/README.md:3]: Ah, my bad. It is not included anywhere.

File: website/src/pages/index.js (hunk 4 above)
Comment (asafkorem) [reply website/src/pages/index.js]: this text is invisible, anyway - removed.

File: website/src/pages/markdown-page.md (hunk 5 above)
Comment (asafkorem) [reply website/src/pages/markdown-page.md]: Indeed. Missed it.
Removed.

File: docs/APIRef.Artifacts.md (hunk 6 above)
Comment (asafkorem) [reply docs/APIRef.Artifacts.md]: Done as discussed.

**Comment by asafkorem** [comment]
> @asafkorem as for the images – could you explicitly list their URLs (here at least), where do they come from, and what their respective licenses are? We just need to keep in mind that we're a public project and need to comply with the copyright policies. Let's be 100% transparent about the assets.

Great point. All photos were taken from [iconarchive.com](https://iconarchive.com/) (and are free for non-commercial use). I've added all references in the comments:
https://github.com/wix/Detox/blob/778ea4482f9aa38dbaad2dc0f776a4b9c7e448d0/website/src/components/HomepageFeatures.js

**Pull request merged**